src/Compression/CompressionCodecDeflate.cpp

```diff
@@ -0,0 +1,357 @@
+    job_ptr->flags = QPL_FLAG_FIRST | QPL_FLAG_DYNAMIC_HUFFMAN | QPL_FLAG_LAST | QPL_FLAG_OMIT_VERIFY;
+    status = qpl_execute_job(job_ptr);
+    if (QPL_STS_OK == status)
+    {
+        compressed_size = job_ptr->total_out;
+    }
+    else
+        LOG_WARNING(log, "HardwareCodecDeflate::doCompressData fail ->status: '{}' ", static_cast<size_t>(status));
+
+    DeflateJobHWPool::instance().releaseJob(job_id);
+    return compressed_size;
+}
+
+uint32_t HardwareCodecDeflate::doDecompressData(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size) const
+{
+    uint32_t job_id = 0;
+    qpl_job * job_ptr = DeflateJobHWPool::instance().acquireJob(&job_id);
+    if (job_ptr == nullptr)
+    {
+        LOG_WARNING(log, "HardwareCodecDeflate::doDecompressData acquireJob fail!");
+        return 0;
+    }
+    qpl_status status;
+
+    // Performing a decompression operation
+    job_ptr->op = qpl_op_decompress;
+    job_ptr->next_in_ptr = reinterpret_cast<uint8_t *>(const_cast<char *>(source));
+    job_ptr->next_out_ptr = reinterpret_cast<uint8_t *>(dest);
+    job_ptr->available_in = source_size;
+    job_ptr->available_out = uncompressed_size;
+    job_ptr->flags = QPL_FLAG_FIRST | QPL_FLAG_LAST;
+
+    // Decompression
+    status = qpl_execute_job(job_ptr);
+
+    if (status == QPL_STS_OK)
+    {
+        DeflateJobHWPool::instance().releaseJob(job_id);
+        return job_ptr->total_out;
+    }
+    else
+    {
+        LOG_WARNING(log, "HardwareCodecDeflate::doDecompressData fail ->status: '{}' ", static_cast<size_t>(status));
+        DeflateJobHWPool::instance().releaseJob(job_id);
+        return 0;
+    }
+}
+
+uint32_t HardwareCodecDeflate::doDecompressDataReq(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size)
+{
+    uint32_t job_id = 0;
+    qpl_job * job_ptr = DeflateJobHWPool::instance().acquireJob(&job_id);
+    if (job_ptr == nullptr)
+    {
+        LOG_WARNING(log, "HardwareCodecDeflate::doDecompressDataReq acquireJob fail!");
+        return 0;
+    }
+    qpl_status status;
+
+    // Performing a decompression operation
+    job_ptr->op = qpl_op_decompress;
+    job_ptr->next_in_ptr = reinterpret_cast<uint8_t *>(const_cast<char *>(source));
+    job_ptr->next_out_ptr = reinterpret_cast<uint8_t *>(dest);
+    job_ptr->available_in = source_size;
+    job_ptr->available_out = uncompressed_size;
+    job_ptr->flags = QPL_FLAG_FIRST | QPL_FLAG_LAST;
+
+    // Decompression
+    status = qpl_submit_job(job_ptr);
+    if (QPL_STS_OK == status)
+    {
+        jobDecompAsyncMap.insert(std::make_pair(job_id, job_ptr));
+        return job_id;
+    }
+    else
+    {
+        DeflateJobHWPool::instance().releaseJob(job_id);
+        LOG_WARNING(log, "HardwareCodecDeflate::doDecompressDataReq fail ->status: '{}' ", static_cast<size_t>(status));
+        return 0;
+    }
+}
+
+void HardwareCodecDeflate::doDecompressDataFlush()
+{
+    uint32_t job_id = 0;
```

Comment:
I think some high-level source code comments what this method does would be helpful, including l. 191.

Reply:
done